docs/design-proposals/arch/daemonset.md

```diff
@@ -0,0 +1,71 @@
+
+Providing a centralized DaemonSet management tool on top of all and every TP is convenience gain for cluster admin; however it is beyond the core design. The rudimentary utility using kubectl config context is able to mitigate before a better solution is available.
+
+#### Alternatives
+We considered other options for DaemonSet in scale-out Arktos.
+
+**DaemonSet in Resource Partition**
+
+DaemonSet resource is managed combinely by all resource partitions; tenant partition is not involved. Each resource partition has its own copy of the DaemonSet resource; its DaemonSet controller and scheduler manage the DaemonSet in (local) scope of its Nodes.
+
+This architecture has some defects:
+* Lack of built-in mechanism to distribute and sync DaemonSet copies to all tenant partitions;
+* Increased operation complexity: multiple RP's each runs master controllers;
+* Decreased scheudule accuracy: scheduler in RP schedules daemon Pods which cannot have the resource allocation reflected in schedule in TP, which leads to inaccuracy of the deceived resources.
+* Overlap of TP/RP: Pod/configmap/secret may be managed by TP or RP - depending on whether they are DaemonSet related or not, which tarnishes the clear cut of TP/RP.
+
+**DaemonSet in Hybrid**
+
+This is a variant of the above-mentioned DS in RP: cluster admin manages DaemonSet at tenant partition; system implements a built-in distribution mechanism (DaemonSet distribution controller) to push DeamonSet change to all resource partitions.
+
+This architecture mitigates the DS sync problem, but it brings more incompatibility of scale-out and scale-up, as scale-up hs no such sync in the first place.
+
+### Reequired Changes
+The minimum changes required to support DaemonSet of Arktos
+
+#### API Server (admission Controller)
+* The admission controller module to block non-system tenanted DaemonSet.
+* API Servers (at least of tenant partitions) enables the admission controller when it is started
+
+Below is the diagram of DaemonSet admission:
+<p align="center"> <img src="images/daemonset_OPD/daemonset-support-DS-creating-DSadding.jpg"> </p>
+
+
+#### Kubelet
+Kubelet shall be able to identify the API resources (configmap, secret, PVC etc) from the same tenant partition as the daemon Pod to be started, in environment preparing process.
+
+Below is the diagram of kubelet environment preparing process:
+<p align="center"> <img src="images/daemonset_OPD/daemonset-support-DS-creating-podStarting-envPreparing.jpg"> </p>
+
+There are two significances for kubelet, at high level:
```

Comment:
What do you meant by "significances"?

Reply:
will reword it to "significant required functionalities"